internal/jujuapi/controllerroot.go

```diff
@@ -51,6 +51,7 @@ type JIMM interface {
 	GetCloudCredential(ctx context.Context, user *openfga.User, tag names.CloudCredentialTag) (*dbmodel.CloudCredential, error)
 	GetCloudCredentialAttributes(ctx context.Context, u *openfga.User, cred *dbmodel.CloudCredential, hidden bool) (attrs map[string]string, redacted []string, err error)
 	GetCredentialStore() credentials.CredentialStore
+	GetRoleManager() jimm.RoleManager
```

Comment:
Ah now we've walked into an interesting issue here. I don't think this is what we want, but I guess you can't add the methods here because the JIMM struct doesn't implement any role related methods. Maybe embed the RoleManager on JIMM then you can put actual methods here instead.

Reply:
For auth service, credentialstore we've resorted to this `GetThing()` method, so I've c/ped.

@ale8k what do you think?

Reply:
Yeah.. Not keen on this either really but we use a JIMM interface - I think it's a step up than calling methods directly though and the contract is still honoured. I'd say do this for now then let's discuss the humongous JIMM interface?

Reply:
It's easily resolved if we just embed the RoleManger on JIMM. I'll make a PR for that to illustrate then this PR will be easier to land.

Reply:
On second thought, what if we don't export RoleManager and change `GetRoleManager` to just `RoleManager` then the code reads a bit more cleanly as `r.jimm.RoleManager().RemoveRole`. Probably easiest for now.

In fact, if everything was inside a manager, we wouldn't need a large JIMM interface. The interface is done at the level of the manager. I.e. you can still unit test the api layer, you just need a mock manager. I think going all in on the managers will be helpful.

Reply:
we can't use a field and a method with the same name.
we already have `RoleManager` taken, do you have a better name than `GetRoleManager()`?

Reply:
One way is to have `roleManager` private and then having `RoleManager` as the getter and `SetManager` as the setter, but i don't know if that's any better.

Reply:
We can discuss this in standup.